components/api/src/main/java/com/hotels/styx/api/FullHttpRequest.java

```diff
@@ -55,7 +55,34 @@
 import static java.util.stream.Stream.concat;
 
 /**
- * HTTP request with a fully aggregated/decoded body.
+ * An immutable HTTP request object containing a finite size content body in full.
+ * <p>
+ * A {@code FullHttpRequest} is useful for HTTP request messages with
```

Comment:
This seems like a bit of a run-on sentence and not completely clear. Maybe it would be better to write something like "A {@code FullHttpRequest} is useful for HTTP request messages with finite body content, such as when PUT or POST are used to create or modify a RESTful object".

Reply:
Okay. Sounds good.

It now reads:

```
/**
 * An immutable HTTP request object including full body content.
 * <p>
 * A {@link FullHttpRequest} is useful for HTTP request messages with
 * finite body content, such as when PUT or POST are used to create or
 * modify a RESTful object.
 * <p>
 * .... 
```